### PR TITLE
feat(badger-api): Fetch latest Sett snapshots from DynamoDB

### PR DESCRIPTION
**Pull request by ethkaneki**

### CODE REVIEW

File: src/setts/setts.service.ts [hunk 1]
```diff
@@ -82,27 +86,46 @@ export class SettsService {
       sources: [],
     };
 
-    const [settSnapshots, settData] = await Promise.all([
-      this.getSettSnapshots(settName, SAMPLE_DAYS),
-      getSett(chain.graphUrl, settDefinition.settToken),
-    ]);
+    const record = await query({
```
Comment (axejintao) [inline src/setts/setts.service.ts:89]: can we encapsulate this into a helper within the util file? i can see this being useful elsewhere probably.

File: src/setts/setts.service.ts [hunk 2]
```diff
@@ -82,27 +86,46 @@ export class SettsService {
...
+    });
 
-    // set to current balance, fallback to snapshot
+    // Set to current balance, fallback to snapshot
     let balance = 0;
-    if (settData.sett) {
-      const currentSett = settData.sett;
-      balance = currentSett.balance;
-      sett.ppfs = currentSett.balance / currentSett.totalSupply;
+    let settSnapshots: SettSnapshot[] = [];
```
Comment (axejintao) [inline src/setts/setts.service.ts:102]: the sett snapshots are also used for calculating the sampled apy based on historic ppfs delta, so we want to replace the logic that is poulating `settData` which is the "live lookup" vs. replacing the historic data.

Comment (axejintao) [reply src/setts/setts.service.ts:102]: to try to clear it up basically:

sett snapshots = the historic 30 min sampled values for estimating native apy,
sett data = the current state of the sett

so these cached ones need to replace the second as the apy calculation are assuming a set of data at 30 min polling intervals

**Review by axejintao — CHANGES_REQUESTED** [review]

File: src/setts/setts.service.ts (hunk 2 above)
Comment (ethkaneki) [reply src/setts/setts.service.ts]: Discussed offline 👍

File: src/setts/setts.service.ts (hunk 1 above)
Comment (ethkaneki) [reply src/setts/setts.service.ts]: Discussed offline - will move to an AWS / Dynamo service in later PRs

**Review by axejintao — APPROVED** [review]

**Pull request merged**